display.py

```diff
@@ -1,57 +1,124 @@
-        super().__init__()
-        self._font = ImageFont.truetype("DejaVuSans.ttf", 12)
-        self._font_height = self._font.getsize("M")[1]
-        self.rows = display.height // self._font_height
-        self.highlight = -1
-        self._text_lines: list[str] = []
-        self._display_device = display
-        self._display_device.show()
-
-    def clear(self, layer: Literal["text", "top", "all"] = "text"):
-        if layer == "all" or layer == "text":
-            self._display_device.clear()
-            self._text_lines.clear()
-
-    def push_back(self, text: str):
-        redraw = False
-        if len(self._text_lines) >= self.rows:
-            self._text_lines.pop(0)
-            redraw = True
-        self._text_lines.append(text)
-        # TODO: only redraw when redraw is true if false update part of screen
-        self._redraw()
-
-    def push_front(self, text: str):
-        if len(self._text_lines) >= self.rows:
-            self._text_lines.pop()
-        self._text_lines.insert(0, text)
-        self._redraw()
-
-    def highlight_text(self, row: int):
-        self.highlight = row
-        self._redraw() # TODO: redraw only rows that get/stop being highlighted
-
-    def _redraw(self):
-        self._display_device.clear()
-        with canvas(self._display_device) as draw:
-            for index, line in enumerate(self._text_lines):
-                fill = "white"
-                if index == self.highlight:
-                    fill = "black"
-                    draw.rectangle((
-                        0, index * self._font_height,
-                        self._display_device.width, (index + 1) * self._font_height), outline="white", fill="white")
-                draw.text((0, index * self._font_height), line, fill=fill, font=self._font)
-
-    def update_row(self, row: int, text: str):
-        self._text_lines[row] = text
-        self._redraw() # TODO: redraw only updated row
\ No newline at end of file
+
+class Terminal(luma_terminal):
+    def __init__(self, device: backlit_device, font=None, color="white", bgcolor="blue", tabstop=4, line_height=None, animate=False, word_wrap=False):
+        # if depth > 0 then don't flush(don't display on device just write to bitmap/image)
+        self.context_manager_depth = 0
+        self.scroll = False
+        super().__init__(device, font, color, bgcolor, tabstop, line_height, animate, word_wrap)
+
+    def goto(self, x: int, y: int) -> None:
+        if (0 <= x < self.width) and (0 <= y < self.height):
+            #pylint: disable=attribute-defined-outside-init
+            self._cx = self._cw * x
+            self._cy = self._ch * y
+            #pylint: enable=attribute-defined-outside-init
+
+    @property
+    def x(self):
+        return self._cx // self._cw
+
+    @property
+    def y(self):
+        return self._cy // self._ch
+
+    def println(self, text="", *, highlight = False, fill = True, scroll_first = False) -> None:

```

Comment:
Please align convention -> `highlight=False` and follow.
can you please help me understand why do we have this `*` there?

Reply:
Mostly because println("test", True, True, True) isn't readable. Or even println("test", True). With * I have to use argument names. Less chance to make a mistake.

Reply:
Makes sense, didn't know about that. Thanks!